.circleci/config.yml

```diff
@@ -2,24 +2,65 @@ version: 2.1
 orbs:
   cypress: cypress-io/cypress@1
   codecov: codecov/codecov@1.0.4
+  aws-s3: circleci/aws-s3@1.0.11
+
 jobs:
+  build:
+    docker: 
+      - image: circleci/node:10
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - npm-{{ checksum "package-lock.json" }}
+            - npm-
+      - run:
+          name: Set registry URL
+          command: 'npm config set "@fortawesome:registry" https://npm.fontawesome.com/'
+      - run:
+          name: Authenticate with registry
+          command: 'npm config set "//npm.fontawesome.com/:_authToken" $FONTAWESOME'
+      - run:
+          name: Install
+          command: npm i --no-dev
+      - run: 
+          name: Run build script
+          command: 'npm run build'
+      - save_cache:
+          key: npm-{{ checksum "composer.lock" }}
+          paths:
+            - ./node_modules
+      - run:
+          name: Create built_code directory
+          command: mkdir -p ./built_code/
+      - run:
+          name: move dist directory into built_code directory
+          command: mv ./dist built_code/
+      - persist_to_workspace:
+          root: ./built_code
+          paths:
+            - dist
   deploy:
-    executor: cypress/base-10
+    docker:
+      - image: 'circleci/python:2.7'
     steps:
       - attach_workspace:
-          at: ~/
-      - run: npm run build
-      - run: echo S3 orb stuff for dist/
```

Comment:
I _kinda_ think you could drop all of the build steps added and just add this at line 12:
```yml
  - aws-s3/sync:
      from: /dist
      to: s3://$bucket_name
      overwrite: true
```

Reply:
If we reuse built items, most likely, yes.

Reply:
well I mean the build runs on line 11.  I don't see a need to create a new environment to do the build, the results from line 11 and the new build should be identical.

Reply:
well I mean the build runs on line 11.  I don't see a need to create a new environment to do the build.. the results from line 11 and the new build should be identical.

Reply:
👍 Alsot, unforunately, the aws-s3 orb from CircleCI only includes commands and not jobs. So, it still needs to be run in a defined job context.